### PR TITLE
Added UserClaimsPrincipal to ViewComponent 

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #4964 

@Eilon

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs [hunk 1]
```diff
@@ -61,6 +62,17 @@ public IPrincipal User
         }
 
         /// <summary>
+        /// Gets the <see cref="ClaimsPrincipal"/> for the current user.
+        /// </summary>
+        public ClaimsPrincipal UserClaimsPrincipal
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs]: `public ClaimsPrincipal UserClaimsPrincipal => ViewContext?.HttpContext?.User` for 💯 points!

Comment (jbagga) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs]: Then it won't be consistent with the others. Is that okay?

Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs]: @jbagga no, it'll be better! 😄

**Comment by jbagga** [comment]
@Eilon @pranavkm Updated

**Comment by pranavkm** [comment]
:shipit: Just make sure there's exactly one commit going in for this change (The commit history looks suspect).

**Pull request merged**